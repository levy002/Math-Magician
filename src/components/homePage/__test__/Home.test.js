import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from '../Home';

test('renders "The history of mathematics" on homepage link', () => {
  render(<Home />);
  const linkElement = screen.getByText(/The history of mathematics/i);
  expect(linkElement).toBeTruthy();
});

it('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
