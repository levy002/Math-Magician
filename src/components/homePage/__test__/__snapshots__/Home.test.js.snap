// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="history"
>
  <h1
    className="title"
  >
    HISTORY
  </h1>
  <p>
    The history of mathematics deals with the origin of discoveries in mathematics and the mathematical methods and notation of the past. Before the modern age and the worldwide spread of knowledge, written examples of new mathematical developments have come to light only in a few locales. From 3000 BC the Mesopotamian states of Sumer, Akkad and Assyria, followed closely by Ancient Egypt and the Levantine state of Ebla began using arithmetic, algebra and geometry for purposes of taxation, commerce, trade and also in the patterns in nature, the field of astronomy and to record time and formulate calendars.
  </p>
  <p>
    The earliest mathematical texts available are from Mesopotamia and Egypt – Plimpton 322 (Babylonian c. 2000 – 1900 BC), the Rhind Mathematical Papyrus (Egyptian c. 1800 BC) and the Moscow Mathematical Papyrus (Egyptian c. 1890 BC). All of these texts mention the so-called Pythagorean triples, so, by inference, the Pythagorean theorem seems to be the most ancient and widespread mathematical development after basic arithmetic and geometry.The study of mathematics as a demonstrative discipline began in the 6th century BC with the Pythagoreans, who coined the term mathematics from the ancient Greek μάθημα (mathema), meaning subject of instruction\\. Greek mathematics greatly refined the methods (especially through the introduction of deductive reasoning and mathematical rigor in proofs) and expanded the subject matter of mathematics.
  </p>
  <p
    className="reference"
  >
    Reference:
    <a
      href="https://en.wikipedia.org/wiki/History_of_mathematics"
    >
      Wikipedia
    </a>
  </p>
</div>
`;
